pkg/ebpf/c/include/tracee/context.h

```diff
@@ -27,13 +34,15 @@ init_context(void *ctx, event_context_t *context, struct task_struct *task, u32
     context->task.pid_id = get_task_pid_ns_id(task);
     context->task.uid = bpf_get_current_uid_gid();
     context->task.flags = 0;
+
     if (is_compat(task))
         context->task.flags |= IS_COMPAT_FLAG;
+
     __builtin_memset(context->task.comm, 0, sizeof(context->task.comm));
+
     ret = bpf_get_current_comm(&context->task.comm, sizeof(context->task.comm));
     if (unlikely(ret < 0)) {
-        // disable logging as a workaround for instruction limit verifier error on kernel 4.19
-        // tracee_log(ctx, BPF_LOG_LVL_ERROR, BPF_LOG_ID_GET_CURRENT_COMM, ret);
+        tracee_log(ctx, BPF_LOG_LVL_ERROR, BPF_LOG_ID_GET_CURRENT_COMM, ret);
```

Comment:
This will only work on kernel 5.2 and above.
We still didn't take the decision to only support these kernels and up (when we will, we should also update some loop limits and more)

Reply:
That's also true for many parts of the rest of eBPF code. Take networking code for example, it works:

- 4.18 RHEL kernels
- >= 5.2 kernels

We never consider 5.0, 5.1 kernels because there are no distros using those as LTS kernels. Why is this different ? I believe if we are dropping non CORE we should stick with platforms we DO support, and those kernel versions (what environment do you know use 5.1 kernel that we do support ?).

Reply:
Network code is the exception. I remind you that we originally thought the cgroup logic will work on older kernels as well, but then discovered some helpers were missing. We never took the decision to drop support for older kernels for the other features (maybe we should, but this should be open for discussion first)

Reply:
My point is, how can we "worry" about supporting kernels that aren't supported upstream anymore ? (Distros or Upstream LTS tree).

That is why I believe we should have a "supported kernels/environments" list (and that is, IMHO, needed for dropping non CORE).

Example: kernel 5.1 DOES have a verifier issue that brakes Tracee for a few cases, but it will never be fixed (no distro or no upstream stable tree supports it).

Should we say we support 5.1 ? I don't think we could even if we wanted.

Reply:
I'm not saying we should support 5.1 kernels, just that the current prerequisites (in the documentation) to run tracee is minimal kernel 4.18, with the exception of network functionality. If we want to change this prerequisite, we need to do a separate discussion, and not just remove support as part of dropping non-core

Reply:
And I'm saying the discussion would be empty because CORE needs either 4.18 RHEL8 or 5.4, 5.8, 5.10, 5.13, 5.15, 5.19, 6.1 kernels from distros/upstream (that are either supported upstream OR by a distro).

So I asked you what kernels you think of support other than those, the "core" part of the discussion is here.

There is no way we can support Tracee in an unsupported kernel (no matter the discussion we want to have) because the nature of a possible bug might not be in Tracee itself (and that would leave no possible way for us to fix "the issue").

The prerequisite is wrong.